crates/c-api/include/wasmtime.h

```diff
@@ -38,6 +38,12 @@ WASMTIME_CONFIG_PROP(cranelift_opt_level, wasmtime_opt_level_t)
 
 ///////////////////////////////////////////////////////////////////////////////
 
+bool wasmtime_wat2wasm(
+    wasm_byte_vec_t *ret,
+    wasm_engine_t *engine,
+    const char* wat,
+);
```

Comment:
The `ret` parameter looks out of order in comparison with rest of C API. Closest function I can find is [wasm_module_serialize](https://github.com/WebAssembly/wasm-c-api/blob/master/include/wasm.h#L402). Can we adjust signature to `bool wasmtime_wat2wasm(wasm_engine_t *engine, const char* wat, own wasm_byte_vec_t *ret)`. Do we also want to make the wat parameter as `const wasm_byte_vec_t*` similar to wasm_module_new?

Reply:
Makes sense to reorder the out parameter, and yeah looking at the C API it looks like similar-functions (e.g. `wasm_importtype_new`) take a `wasm_byte_vec_t` parameter, so makes sense to do that here too.